### ISSUE TITLE
Weird Android Path Rendering

### ISSUE THREAD

**Issue by administerspice** [react-native-community/react-native-svg#325]
"react": "16.0.0-alpha.6",
    "react-native": "0.44.0",
    "react-native-svg": "https://github.com/react-native-community/react-native-svg"

I have path that is supposed to be a 270 degree arc. It looks great on iOS but is not round on Android...
`    let pathCommands = "M" + M_X + "," + M_Y +
                        "A" + RADIUS + "," + RADIUS + " 270 1 1 " + E_X + "," + E_Y`
        
`<Path d={pathCommands}
              fill="transparent"
              stroke="url(#linear-gradient)"
              strokeWidth={STROKE_WIDTH}
              strokeLinecap="round"
            />`

ANDROID:
![screenshot_20170502-171837](https://cloud.githubusercontent.com/assets/8702001/25639935/c97df392-2f5b-11e7-8a24-c58d4ed7e30e.png)

iOS:
![screen shot 2017-05-02 at 5 23 57 pm](https://cloud.githubusercontent.com/assets/8702001/25639999/f62a85ea-2f5b-11e7-8b1e-1d8986f7ffe5.png)

**Comment by magicismight** [comment]
Thanks for reporting this.
Can you print the `d` prop?

**Comment by wildQueequeg** [comment]
@magicismight The size of the arc is dynamic, so the radius, beginning and end points are also dynamic. d={pathCommands}

const M_X = (CENTER_X) + (RADIUS*Math.cos(ANGLE_RADIANS_START))
const M_Y = (CENTER_Y) + (RADIUS*Math.sin(ANGLE_RADIANS_START))
const E_X = (CENTER_X) + (RADIUS*Math.cos(ANGLE_RADIANS_END))
const E_Y = (CENTER_Y) + (RADIUS*Math.sin(ANGLE_RADIANS_END))

let pathCommands = "M" + M_X + "," + M_Y + "A" + RADIUS + "," + RADIUS + " 270 1 1 " + E_X + "," + E_Y

**Comment by wildQueequeg** [comment]
@magicismight, any idea what is causing this?

**Comment by msand** [comment]
Are you still having this issue? If you are, then it's probably the path parser in the native code part.

**Comment by jayhasyee** [comment]
I have the same issue. The arcs are a bit angular, however the browser shows well.
react-native-svg@6.3.1
react-native@0.52.0

**Comment by davidpricedev** [comment]
I'm seeing this as well. Here is the code that reproduces the issue:

html version - makes a nice oval: 
```html
<div style="width: 270px; height: 480px;">
    <svg viewbox="0 0 270 480">
        <path d="
          M 0 0 H270 V480 H0z
          M 135 240 m -105, 0
          a 2,3.8 0 1,0 210,0
          a 2,3.8 0 1,0 -210,0z"/>
    </svg>
</div>
```

react-native + react-native-svg on android - a strange egg-shaped oval: 
```javascript
<View style={{
                position: 'absolute',
                left: 0,
                right: 0,
                top: 0,
                right: 0,
                alignItems: 'center',
                justifyContent: 'center'  }}>
    <Svg viewbox={`0 0 270 480`} width={270} height={480}>
        <Path d={`
                    M 0 0 H270 V480 H0z
                    M 135 240 m -105, 0
                    a 2,3.8 0 1,0 210,0
                    a 2,3.8 0 1,0 -210,0z`} />
    </Svg>
</View>
```

I can post this as a repository that reproduces the issue too if that would help.

Here is what it looks like on android:
![screen shot 2018-06-01 at 13 03 25](https://user-images.githubusercontent.com/6391952/40856098-36874796-659c-11e8-8c25-9fd7777347e5.png)

**Comment by davidpricedev** [comment]
I've had a look through the ios vs android sides for the arc drawing and they look pretty similar - up to the point where they hand off to the native bezier/path code. However, when I looked at what is happening at runtime - there is quite a difference.

The first thing that is noticeable is that this generates 5 calls to `arcToBezier` on the android side, but only 4 on the ios side. The last 3 calls on each platform seem to be very similar and have the same inputs into the `arcToBezier` function, so I'll assume they should be the same.

The second thing is the `n` value in `arcToBezier` seems to be the crux of the difference. On the second call in ios vs the 3rd call in android, ios has the n value as 2 and android has it at 3. In digging into what it should be I have this: 

Relevant line of code from java: 
`int n = (int) Math.ceil(Math.abs(arc / (Math.PI / 2)));`

## Correct steps (what I can do on paper): 

1. `Math.PI / 2` = 1.57 or PI/2
1. `-PI / (PI/2)` = -2 
1. `Math.abs(-2)` = 2 
1. `int n = (int) Math.ceil(2)` = 2

## Android case (what happens in the code): 

1. `Math.PI / 2` = 1.57 or PI/2
1. `-PI / (PI/2)` = -2.0000000556550703
1. `Math.abs(-2.0000000556550703)` = 2.0000000556550703 
1. `int n = (int) Math.ceil(2.0000000556550703)` = 3

It looks like this is a classic case of floating point math. One solution would be to round the inner result: 

```
- int n = (int) Math.ceil(Math.abs(arc / (Math.PI / 2)));
+ double arcOverHalfPi = arc / (Math.PI / 2);
+ // no round-to-decimal-places function in java so we have to multiply and divide by 10^n to get n decimal places
+ double roundedArcOverHalfPi = Math.round(arcOverHalfPi * 1000) / 1000D;
+ int n = (int) Math.ceil(Math.abs(roundedArcOverHalfPi));
```

That fixes the egg shape. The peaks are still too sharp on android compared to iOS though.
![screen shot 2018-06-01 at 17 04 23](https://user-images.githubusercontent.com/6391952/40865439-e0d20800-65bd-11e8-9c52-4b8627032837.png)

**Comment by alexsmartens** [comment]
Hi @administerspice, you have a really cool screenshot in your initial post. I'm looking to do something similar. Don't you mind sharing the snippet of your code showing how you did it?

**Comment by msand** [comment]
Fixed in #704
